File: src/main/java/com/wesker926/leetcode/algorithms/p1446/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/12/1
// @description: 遍历
package s1

func maxPower(s string) int {
	max, cur := 1, 1
	for i := 1; i < len(s); i++ {
		if s[i] == s[i-1] {
			cur++
		} else {
			max = maxInt(max, cur)
			cur = 1
		}
	}
	return maxInt(max, cur)
}

func maxInt(i, j int) int {
	if i >= j {
		return i
	}
	return j
}

File: src/main/java/com/wesker926/leetcode/algorithms/p1446/s1/Solution.java
package com.wesker926.leetcode.algorithms.p1446.s1;

/**
 * @author wesker.gh
 * @date 2021/12/1
 * @description 遍历
 */
public class Solution {
    public int maxPower(String s) {
        int max = 1, cur = 1;
        char[] cs = s.toCharArray();
        for (int i = 1; i < cs.length; i++) {
            if (cs[i] == cs[i - 1]) {
                cur++;
            } else {
                max = Math.max(max, cur);
                cur = 1;
            }
        }
        return Math.max(max, cur);
    }
}
